lib/cbdocument.utils.js

```diff
@@ -131,40 +131,82 @@ exports.buildRefValues = function (indexes, data) {
       const path = indexes[v].path;
       const name = indexes[v].name;
       const indexType = indexes[v].indexType;
-      const value = mpath.get(path, data);
-      let valToStore = null;
-      if (typeof value === 'string') {
-        valToStore = value;
-      }
-      if (typeof value === 'number') {
-        valToStore = value.toString();
-      }
-      if (typeof value === 'object' && value._isBaseModel) {
-        valToStore = value.getDocumentKeyValue(value.config.storeFullReferenceId);
-      } else if (Array.isArray(value)) {
-        valToStore = _.map(value, v => {
-          if (v) {
-            if (typeof v === 'string') {
-              return v;
-            }
-            if (typeof v === 'number') {
-              return v.toString();
-            }
-            if (typeof v === 'object' && v._isBaseModel) {
-              return v.getDocumentKeyValue(v.config.storeFullReferenceId);
+
+      const processValue = initialValue => {
+        let finalVal = null;
+        if (typeof initialValue === 'string') {
+          finalVal = initialValue;
+        }
+        if (typeof initialValue === 'number') {
+          finalVal = initialValue.toString();
+        }
+        if (typeof initialValue === 'object' && initialValue._isBaseModel) {
+          finalVal = initialValue.getDocumentKeyValue(initialValue.config.storeFullReferenceId);
+        } else if (Array.isArray(initialValue)) {
+          finalVal = _.map(initialValue, v => {
+            if (v) {
+              if (typeof v === 'string') {
+                return v;
+              }
+              if (typeof v === 'number') {
+                return v.toString();
+              }
+              if (typeof v === 'object' && v._isBaseModel) {
+                return v.getDocumentKeyValue(v.config.storeFullReferenceId);
+              }
             }
-          }
+          });
+
+          finalVal = _.compact(finalVal);
+          finalVal = finalVal.sort();
+        } else if (_.isUndefined(initialValue) || _.isNull(initialValue)) {
+          finalVal = null;
+        }
+        return finalVal;
+      }
+
+      let value = null;
+      if (indexes[v].compound) {
+        value = _.map(path, p => {
+          return processValue(mpath.get(p, data));
         });
 
-        valToStore = _.compact(valToStore);
-        valToStore = valToStore.sort();
-      } else if (_.isUndefined(value) || _.isNull(value)) {
-        valToStore = null;
+        const expandValues = valueList => {
```

Comment:
same here if there is no reason for it to be within this function, make it separate.